backend/asmgen.ml

```diff
@@ -448,7 +464,12 @@ let compile_phrases ~ppf_dump ps =
           if !dump_cmm then fprintf ppf_dump "%a@." Printcmm.phrase p;
           match p with
           | Cfunction fd ->
-            compile_fundecl ~ppf_dump ~funcnames fd;
+            let compile_fundec = compile_fundecl ~ppf_dump ~funcnames in
+            (* Only profile if function level granularity selected*)
+            let profiled_compile_fundec = match !profile_granularity with
+            | Function_level -> Profile.record ~accumulate:true fd.fun_name.sym_name compile_fundec
+            | File_level -> compile_fundec
+            in profiled_compile_fundec fd;
```

Comment:
This is a bit hard to read, maybe just improve formatting or don't let bind `profiled_compile_fundec`.

Reply:
Replaced with [this](https://github.com/ocaml-flambda/flambda-backend/blob/177ba48b5fa25f618ecedc49e05f7d3790dc4626/backend/asmgen.ml#L468).